client/hotrod-client/src/main/java/org/infinispan/client/hotrod/configuration/ConfigurationBuilder.java

```diff
@@ -330,8 +330,8 @@ public ConfigurationBuilder uri(URI uri) {
 
    @Override
    public ConfigurationBuilder uri(String uri) {
-      this.read(HotRodURI.create(uri).toConfigurationBuilder().build(false));
-      return this;
+      HotRodURI hotRodURI = HotRodURI.create(uri);
```

Comment:
I'm afraid this is wrong: it will reset the configuration builder to the contents of the URI.

Reply:
The test shows that no: https://github.com/infinispan/infinispan/pull/9175/files#diff-a1e5b9f2075f0be3c5ee7940703ca7c4730783f723e332f49cc4bc479ed4f060R651

Reply:
If you would like to say `overwrite`. I think it is desirable. But we don't have a test for that

Reply:
Also, if you configure something via API and have a `hotrod-client.properties` file. The `hotrod-client.properties` will overwrite the configuration via API.

Reply:
It took some hours to see what you mean
```java
   public ConfigurationBuilder toConfigurationBuilder(ConfigurationBuilder builder) {
      for(InetSocketAddress address : addresses) {
         builder.addServer().host(address.getHostString()).port(address.getPort());
      }
      if (ssl) {
         builder.security().ssl().enable();
      }
      if (username != null) {
         builder.security().authentication().username(username);
      }
      if (password != null) {
         builder.security().authentication().password(password);
      }
      builder.withProperties(properties);
      return builder;
   }
```

`builder.withProperties(properties);` is going to reset

Reply:
IMHO, withProperties must not reset. WDYT ?

Reply:
I just saw that the following also do the reset
```java
   @Override
   public ConfigurationBuilder uri(URI uri) {
      this.read(HotRodURI.create(uri).toConfigurationBuilder().build(false));
      return this;
   }

   @Override
   public ConfigurationBuilder uri(String uri) {
      this.read(HotRodURI.create(uri).toConfigurationBuilder().build(false));
      return this;
   }
```

Reply:
My example is also not working because it reset the client intelligence.

```java
         Properties properties = new Properties();
         properties.setProperty(ConfigurationProperties.DEFAULT_EXECUTOR_FACTORY_POOL_SIZE, "1");
         properties.setProperty(ConfigurationProperties.URI, cluster.uri());
         ConfigurationBuilder cb = new ConfigurationBuilder().withProperties(properties);
```

Reply:
The only way that I was able to get my example working was
```
         ConfigurationBuilder cb = HotRodURI.create(cluster.uri()).toConfigurationBuilder();
         Properties properties = new Properties();
         properties.setProperty(ConfigurationProperties.DEFAULT_EXECUTOR_FACTORY_POOL_SIZE, "1");
         cb.asyncExecutorFactory().withExecutorProperties(properties);
```